### PR TITLE
sensor: vl53l0x: add dts support ranging params

### PR DESCRIPTION
**Pull request by wuguanghai45**
add dts option to set range profile

### CODE REVIEW

**Review by yperess — CHANGES_REQUESTED** [review]

File: drivers/sensor/vl53l0x/vl53l0x.c [hunk 1]
```diff
@@ -381,7 +383,10 @@ static int vl53l0x_pm_action(const struct device *dev,
 #define VL53L0X_INIT(inst)						 \
 	static struct vl53l0x_config vl53l0x_##inst##_config = {	 \
 		.i2c = I2C_DT_SPEC_INST_GET(inst),			 \
-		.xshut = GPIO_DT_SPEC_INST_GET_OR(inst, xshut_gpios, {}) \
+		.xshut = GPIO_DT_SPEC_INST_GET_OR(inst, xshut_gpios, {}), \
+		.max_time_for_ranging = DT_INST_PROP_OR(inst, max_time_for_ranging, 33000),\
+		.signal_limit_ratio = DT_INST_PROP_OR(inst, signal_limit_ratio, 100), \
+		.sigma_limit_ratio = DT_INST_PROP_OR(inst, sigma_limit_ratio, 60000) \
```
[the remark this answers is not in the document]
Comment (wuguanghai45) [reply drivers/sensor/vl53l0x/vl53l0x.c]: changed

**Review by yperess — DISMISSED** [review]

**Review by yperess — DISMISSED** [review]

File: dts/bindings/sensor/st,vl53l0x.yaml [hunk 2]
```diff
@@ -12,3 +26,25 @@ properties:
     type: phandle-array
     description: |
       Xshutdown pin, Active LOW.
+
+  max-time-for-ranging:
+    type: int
+    default: 33000
+    description: |
+      Set max time for ranging in μs
```
Comment (avisconti) [inline dts/bindings/sensor/st,vl53l0x.yaml:34]: I think you need to explain the reason for default value.

**Review by avisconti — CHANGES_REQUESTED** [review]
A small request but LGTM otherwise.
Can you also add in the PR description on what board you tested it?

File: dts/bindings/sensor/st,vl53l0x.yaml (hunk 2 above)
Comment (wuguanghai45) [reply dts/bindings/sensor/st,vl53l0x.yaml:34]: ok

Comment (wuguanghai45) [reply dts/bindings/sensor/st,vl53l0x.yaml:34]: resolved

**Review by avisconti — DISMISSED** [review]

File: dts/bindings/sensor/st,vl53l0x.yaml (hunk 2 above)
Comment (MaureenHelm) [reply dts/bindings/sensor/st,vl53l0x.yaml:34]: The "explanation" just refers to a preprocessor macro defined in the driver. It doesn't actually explain why the value was selected. Is it a register reset value in the sensor?

Same comment applies to the additional new properties defined below.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Review by avisconti — APPROVED** [review]

**Review by yperess — APPROVED** [review]

**Comment by nashif** [comment]
@MaureenHelm  your change request addressed?

**Comment by erwango** [comment]
@wuguanghai45 Please rebase

**Pull request closed without merging**